lib/matplotlib/widgets.py

```diff
@@ -2499,6 +2499,217 @@ def _onmove(self, event):
         self.update()
 
 
+class PolygonSelector(_SelectorWidget):
+    """Select a polygon region of an axes.
+
+    Place vertices with each mouse click, and make the selection by completing
+    the polygon (clicking on the first vertex). Hold the *ctrl* key and click
+    and drag a vertex to reposition it (the *ctrl* key is not necessary if the
+    polygon has already be completed). Hold the *shift* key and click and drag
+    anywhere in the axes to move all vertices. Press the *esc* key to start a
+    new polygon.
+
+    For the selector to remain responsive you must keep a reference to
+    it.
+
+    Parameters
+    ----------
+    ax : :class:`~matplotlib.axes.Axes`
+        The parent axes for the widget.
+    onselect : function
+        When a polygon is completed or modified after completion,
+        the `onselect` function is called and passed a list of the vertices as
+        ``(xdata, ydata)`` tuples.
+    useblit : bool, optional
+    lineprops : dict, optional
+        The line for the sides of the polygon is drawn with `lineprops`. The
```

Comment:
Can you give me some more info re: "references do not work to arguments"? I'm not sure what you mean.